src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -787,8 +837,8 @@ MulticopterPositionControl::limit_pos_sp_offset()
 	pos_sp_offs.zero();
 
 	if (_control_mode.flag_control_position_enabled) {
-		pos_sp_offs(0) = (_pos_sp(0) - _pos(0)) / _params.sp_offs_max(0);
-		pos_sp_offs(1) = (_pos_sp(1) - _pos(1)) / _params.sp_offs_max(1);
+		/* Limit maximum setpoint offset in position in body frame */
+		scale_with_body_fixed_gains(pos_sp_offs, _pos_sp - _pos, _params.sp_offs_max.einv(), false);
```

Comment:
The last commit https://github.com/PX4/Firmware/pull/5555/commits/762755af66ced74c08de3292a5d89af4c91c1ee8 actually removed the added protection from inv implementation as requested in the review. The argument was to not alter the numerical value in a mathematical operation. So we should rather have the protection in updating `_params.sp_offs_max` to have a min value greater than 0